DVConstraints.py

```diff
@@ -718,6 +717,162 @@ def addThicknessConstraints1D(self, ptList, nCon, axis,
+            option will use different bounds for each constraint.
+
+        scaled : bool
+            Flag specifying whether or not the constraint is to be
+            implemented in a scaled fashion or not.
+
+            * scaled=True: The initial length of each thickness
+              constraint is defined to be 1.0. In this case, the lower
+              and upper bounds are given in multiple of the initial
+              length. lower=0.85, upper=1.15, would allow for 15%
+              change in each direction from the original length. For
+              aerodynamic shape optimizations, this option is used
+              most often.
+
+            * scaled=False: No scaling is applied and the phyical lengths
+              must be specified for the lower and upper bounds.
+
+        scale : float or array of size nSpan
+            This is the optimization scaling of the
+            constraint. Typically this parameter will not need to be
+            changed. If the thickness constraints are scaled, this
+            already results in well-scaled constraint values, and
+            scale can be left at 1.0. If scaled=False, it may changed
+            to a more suitable value of the resulting physical
+            thickness have magnitudes vastly different than O(1).
+
+        name : str
+            Normally this does not need to be set. Only use this if
+            you have multiple DVCon objects and the constriant names
+            need to be distinguished **or** you are using this set of
+            thickness constraints for something other than a direct
+            constraint in pyOptSparse.
+
+        addToPyOpt : bool
+            Normally this should be left at the default of True. If
+            the values need to be processed (modified) BEFORE they are
+            given to the optimizer, set this flag to False.
+        """
+        self._checkDVGeo()
+
+        # Create mesh of itersections
+        constr_line = pySpline.Curve(X=leList, k=2)
+        s = numpy.linspace(0, 1, nSpan)
+        X = constr_line(s)
+        coords = numpy.zeros((nSpan, 3, 3))
+
+        # Project all the points
+        for i in range(nSpan):
+            # Project actual node:
+            up, down, fail = geo_utils.projectNode(
+                X[i], axis, self.p0, self.v1, self.v2)
+            if fail > 0:
+                raise Error("There was an error projecting a node "
+                            "at (%f, %f, %f) with normal (%f, %f, %f)."% (
+                                X[i, 0], X[i, 1], X[i, 2],
+                                axis[0], axis[1], axis[2]))
+            coords[i, 0] = up
+            coords[i, 1] = down
+
+        # Calculate mid-points
+        midPts = (coords[:,0,:] + coords[:,1,:]) / 2.0
+
+        # Project to get leading edge point
+        lePts = numpy.zeros((nSpan, 3))
+        chordDir /= numpy.linalg.norm(numpy.array(chordDir, 'd'))
```

Comment:
this wont work if chordDir is given as a list. I think you can also set the default to -1 0 0 as an array, I just wanted to say that having the option to modify this is good imo. You can keep where you are setting the value explicitly in the test you wrote